lib/ipmi_fru.c

```diff
@@ -600,16 +610,12 @@ int write_fru_area(struct ipmi_intf *intf, struct fru_info *fru, uint8_t id,
 			break;
 		}
 
-		if (rsp->ccode == IPMI_CC_REQ_DATA_INV_LENGTH
-		    || rsp->ccode == IPMI_CC_REQ_DATA_FIELD_EXCEED
-		    || rsp->ccode == IPMI_CC_CANT_RET_NUM_REQ_BYTES) {
-			if (fru->max_write_size > 8) {
-				fru->max_write_size -= 8;
-				lprintf(LOG_INFO,
-					"Retrying FRU write with request size %d",
-					fru->max_write_size);
-				continue;
-			}
+		if (fru_cc_rq2big(rsp->ccode) && fru->max_write_size > 8) {
+			fru->max_write_size -= 8;
+			lprintf(LOG_INFO,
+				"Retrying FRU write with request size %d",
+				fru->max_write_size);
+			continue;
 		} else if (rsp->ccode == 0x80) {
```

Comment:
I suppose this `0x80` must actually be `FRU_CC_WRITE_PROTECTED` (not yet defined anywhere, must go to ipmi_fru.h). There also must be `FRU_CC_DEVICE_BUSY` (0x81). Those are the two FRU command-specific completion codes.

Reply:
I was going to put those in ipmi_cc.h so that all the return codes, including the command-specific ones could be found in one place, however, you disagree --> why not put them together?